service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyService.java

```diff
@@ -54,16 +62,24 @@ private void checkForTrailingSlashes(ServiceContext serviceContext) {
         Iterator<URI> connectIterator = connectURIs.iterator();
 
         while (acceptIterator.hasNext() && connectIterator.hasNext()) {
-            String acceptPath = acceptIterator.next().getPath();
-            String connectPath = connectIterator.next().getPath();
-            boolean acceptPathEndsInSlash = acceptPath.endsWith("/");
-            boolean connectPathEndsInSlash = connectPath.endsWith("/");
+            URI acceptURI = acceptIterator.next();
+            URI connectURI = connectIterator.next();
+            String acceptPath = acceptURI.getPath();
+            String connectPath = connectURI.getPath();
+            boolean acceptPathIsSlash = "/".equals(acceptPath);
+            boolean connectPathIsSlash = "/".equals(connectPath);
+            boolean acceptPathNoTrailingSlash = !acceptPath.endsWith("/") && acceptPath.length() > 1;
+            boolean connectPathNoTrailingSlash = !connectPath.endsWith("/") && connectPath.length() > 1;
 
-            if (acceptPathEndsInSlash != connectPathEndsInSlash) {
+            if (acceptPathIsSlash && connectPathNoTrailingSlash) {
+                throw new IllegalArgumentException(String.format(ACCEPT_PATH_IS_SLASH_ERROR_MESSAGE,
+                        serviceContext.getServiceName(), acceptURI, connectURI));
+            } else if (acceptPathNoTrailingSlash && connectPathIsSlash) {
+                throw new IllegalArgumentException(String.format(CONNECT_PATH_IS_SLASH_ERROR_MESSAGE,
+                        serviceContext.getServiceName(), acceptURI, connectURI));
+            } else if (acceptPathNoTrailingSlash != connectPathNoTrailingSlash) {
                 throw new IllegalArgumentException(
-                        "Please fix the gateway configuration file for service '" + serviceContext.getServiceName()
-                                + "'. Accept and connect must both either have trailing slashes,"
-                                + " or both must not end in slashes.");
+                        String.format(ACCEPT_CONNECT_ERROR_MESSAGE, serviceContext.getServiceName()));
```

Comment:
@jitsni I tried using the following config with the above implementation:

```
<accept>http://localhost:8018/a</accept>
<connect>http://localhost:8018</connect> 
```
and the following error message is being displayed
**Configure connect URI as http://localhost:8018/ since accept URI http://localhost:8018/a doesn't end with a trailing /**

In this case adding a trailing slash in connect will not change the triggered error since the path is empty and replaced by a trailing slash. The only change to pass the validation will be to add a / in accept, but the error message does not indicate this.